src/Fixer/Whitespace/MethodChainingIndentationFixer.php

```diff
@@ -0,0 +1,172 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        for ($index = 1, $count = count($tokens); $index < $count; ++$index) {
+            if ($tokens[$index]->isGivenKind(T_OBJECT_OPERATOR)) {
+                if ($this->needLineBreak($index - 1, $tokens)) {
+                    $tokens[$index - 1]->setContent($tokens[$index - 1]->getContent().$lineEnding);
+                    --$index;
+                    continue;
+                }
+
+                $prev = $tokens[$index - 1];
+                $currentWhitespaces = $this->getLineBreak($prev->getContent());
+
+                if (null !== $currentWhitespaces) {
+                    $prevMeaningIndex = $tokens->getPrevMeaningfulToken($index);
+                    $rightWhitespaces = $this->getRightIndents($prevMeaningIndex, $tokens);
+
+                    if ($currentWhitespaces !== $rightWhitespaces) {
+                        $prev->setContent($lineEnding.$rightWhitespaces);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return string
+     */
+    private function getRightIndents($index, Tokens $tokens)
+    {
+        $indent = $this->whitespacesConfig->getIndent();
+
+        for ($i = $index; $i >= 0; --$i) {
+            if ($i > 0) {
+                $codeToFindIndents = $tokens->generatePartialCode($i - 1, $i);
+            } else {
+                $codeToFindIndents = $tokens[$i]->getContent();
+            }
+
+            $currentWhitespaces = $this->getLineBreak($codeToFindIndents);
+
+            if (null !== $currentWhitespaces) {
+                if ($tokens[$i + 1]->isGivenKind(T_OBJECT_OPERATOR) || $this->isMultiLineMethod($i, $index, $tokens)) {
+                    return $currentWhitespaces;
+                }
+
+                return $currentWhitespaces.$indent;
+            }
+        }
+
+        return $indent;
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return bool
+     */
+    private function needLineBreak($index, Tokens $tokens)
+    {
+        $prevMeaningful = $tokens->getPrevMeaningfulToken($index);
+        $isComment = false;
+
+        for ($i = $index; $i > $prevMeaningful; --$i) {
+            if ($tokens[$i]->isGivenKind(T_OBJECT_OPERATOR) || null !== $this->getLineBreak($tokens[$i]->getContent())) {
+                return $isComment;
+            }
+
+            if ($tokens[$i]->isComment()) {
+                $isComment = true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $content
+     *
+     * @return string|null
+     */
+    private function getLineBreak($content)
+    {
+        if (preg_match('/\R{1}([ \t]*)$/', $content, $matches)) {
```

Comment:
`1 ==` would be nice, to have real boolean value (especially since output is `false|int`)